From 86d73148c2477f21bbff13904dcf5f23a9a67065 Mon Sep 17 00:00:00 2001
From: INTEL-CAMERA WIP <wip@localhost>
Date: Tue, 11 Aug 2026 14:24:16 +0200
Subject: [PATCH 1/6] BOZZA media: dt-bindings: Add GalaxyCore GC5035

NON INVIABILE: manutentore ed email sono segnaposto, manca il
Signed-off-by, e il binding non e' stato validato perche' dtschema e
yamllint non sono installati su questa macchina.

Add device tree bindings for the GalaxyCore GC5035 5 Mpixel raw image
sensor, modelled on the already accepted galaxycore,gc05a2.yaml.
---
 .../bindings/media/i2c/galaxycore,gc5035.yaml | 109 ++++++++++++++++++
 1 file changed, 109 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/media/i2c/galaxycore,gc5035.yaml

diff --git a/Documentation/devicetree/bindings/media/i2c/galaxycore,gc5035.yaml b/Documentation/devicetree/bindings/media/i2c/galaxycore,gc5035.yaml
new file mode 100644
index 000000000..9476da689
--- /dev/null
+++ b/Documentation/devicetree/bindings/media/i2c/galaxycore,gc5035.yaml
@@ -0,0 +1,109 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/media/i2c/galaxycore,gc5035.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: GalaxyCore gc5035 5M Pixel MIPI CSI-2 sensor
+
+maintainers:
+  - TODO Nome Cognome <email@reale>
+
+description:
+  The gc5035 is a raw image sensor with a MIPI CSI-2 image data interface
+  and an I2C control bus. The output format is raw Bayer.
+
+properties:
+  compatible:
+    const: galaxycore,gc5035
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    maxItems: 1
+
+  avdd-supply: true
+
+  dvdd-supply: true
+
+  dovdd-supply: true
+
+  reset-gpios:
+    description: Reference to the GPIO connected to the RESETB pin.
+    maxItems: 1
+
+  powerdown-gpios:
+    description: Reference to the GPIO connected to the PWDN pin.
+    maxItems: 1
+
+  port:
+    $ref: /schemas/graph.yaml#/$defs/port-base
+    additionalProperties: false
+    description:
+      Output port node, single endpoint describing the CSI-2 transmitter.
+
+    properties:
+      endpoint:
+        $ref: /schemas/media/video-interfaces.yaml#
+        unevaluatedProperties: false
+
+        properties:
+          data-lanes:
+            items:
+              - const: 1
+              - const: 2
+
+          link-frequencies: true
+
+        required:
+          - data-lanes
+          - link-frequencies
+
+    required:
+      - endpoint
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - avdd-supply
+  - dvdd-supply
+  - dovdd-supply
+  - reset-gpios
+  - port
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/gpio/gpio.h>
+
+    i2c {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        sensor@37 {
+            compatible = "galaxycore,gc5035";
+            reg = <0x37>;
+
+            clocks = <&gc5035_clk>;
+
+            reset-gpios = <&pio 21 GPIO_ACTIVE_LOW>;
+
+            avdd-supply = <&gc5035_avdd>;
+            dovdd-supply = <&gc5035_dovdd>;
+            dvdd-supply = <&gc5035_dvdd>;
+
+            port {
+                sensor_out: endpoint {
+                    data-lanes = <1 2>;
+                    /* 22 times the external clock, 19.2 MHz here */
+                    link-frequencies = /bits/ 64 <422400000>;
+                    remote-endpoint = <&csi_in>;
+                };
+            };
+        };
+    };
+
+...
-- 
2.47.3

